platform/qt/src/qmapboxgl.cpp

```diff
@@ -176,80 +199,146 @@ QMapboxGLSettings::QMapboxGLSettings()
 {
 }
 
-QMapboxGLSettings::MapMode QMapboxGLSettings::mapMode() const
-{
-    return m_mapMode;
-}
-
-void QMapboxGLSettings::setMapMode(MapMode mode)
-{
-    m_mapMode = mode;
-}
+/*!
+    Returns the OpenGL context mode. This is specially important when mixing
+    with other GL draw calls.
 
+    By default, it is set to QMapboxGLSettings::SharedGLContext.
+*/
 QMapboxGLSettings::GLContextMode QMapboxGLSettings::contextMode() const
 {
     return m_contextMode;
 }
 
+/*!
+    Sets the OpenGL context \a mode.
+*/
 void QMapboxGLSettings::setContextMode(GLContextMode mode)
 {
     m_contextMode = mode;
 }
 
+/*!
+    Returns the constrain mode. This is used to limit the map to wrap
+    around the globe horizontally.
+
+    By default, it is set to QMapboxGLSettings::ConstrainHeightOnly.
+*/
 QMapboxGLSettings::ConstrainMode QMapboxGLSettings::constrainMode() const
 {
     return m_constrainMode;
 }
 
+/*!
+    Sets the map constrain \a mode.
+*/
 void QMapboxGLSettings::setConstrainMode(ConstrainMode mode)
 {
     m_constrainMode = mode;
 }
 
+/*!
+    Returns the viewport mode. This is used to flip the vertical
+    orientation of the map as some devices may use inverted orientation.
+
+    By default, it is set to QMapboxGLSettings::DefaultViewport.
+*/
 QMapboxGLSettings::ViewportMode QMapboxGLSettings::viewportMode() const
 {
     return m_viewportMode;
 }
 
+/*!
+    Sets the viewport \a mode.
+*/
 void QMapboxGLSettings::setViewportMode(ViewportMode mode)
 {
     m_viewportMode = mode;
 }
 
+/*!
+    Returns the cache database maximum hard size in bytes. The database
+    will grow until the limit is reached. Setting a maximum size smaller
+    than the current size of an existing database is undefined.
```

Comment:
It's appropriate to leave an edge case undefined, to keep our options open in the future. The usual wording is "results in undefined behavior".